collie/model/base/base_pipeline.py

```diff
@@ -722,6 +733,116 @@ def _get_item_embeddings(self) -> torch.tensor:
             '``_get_item_embeddings`` is not implemented in this subclass.'
         )
 
+    def get_user_predictions(self,
+                             item_id: int = 0,
+                             unseen_users_only: bool = False,
+                             sort_values: bool = True) -> pd.Series:
+        """
+        User counterpart to ``get_item_predictions`` method.
+
+        Get predicted rankings/ratings for all users for a given ``item_id``.
+
+        This method cannot be called for datasets stored in ``HDF5InteractionsDataLoader`` since
+        data in this ``DataLoader`` is read in dynamically.
+
+        Parameters
+        ----------
+        item_id: int
+        unseen_users_only: bool
+            Filter ``preds`` to only show predictions of unseen users not present in the training
+            or validation datasets for that ``item_id``. Note this requires both ``train_loader``
+            and ``val_loader`` to be 1) class-level attributes in the model and 2) DataLoaders with
+            ``Interactions`` at its core (not ``HDF5Interactions``). If you are loading in a model,
+            these two attributes will need to be set manually, since datasets are NOT saved when
+            saving the model
+        sort_values: bool
+            Whether to sort recommendations by descending prediction probability
+
+        Returns
+        -------
+        preds: pd.Series
+            Sorted values as predicted ratings for each user in the dataset with the index being
+            the user ID
+        """
+        if item_id > self.hparams.num_items:
+            raise ValueError(
+                f'``item_id`` {item_id} is not in the model. '
+                f'Expected id between 0 and {self.hparams.num_items}'
```

Comment:
I prefer being super transparent about why we are raising this error. If we instead phrase it like this: 

```suggestion
                f'Expected ID between ``0`` and ``self.hparams.num_items``, {self.hparams.num_items}'
```

then the user knows that, if for some reason they want to try it anyway, they just need to do: 

```python
model.hparams.num_items = XX
```

What do you think about this?